components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -458,7 +460,9 @@ class GeckoEngineSession(
                 interceptor != null && (!request.isDirectNavigation || interceptor.interceptsAppInitiatedRequests())
             ) {
                 val engineSession = this@GeckoEngineSession
-                val isSameDomain = engineSession.currentUrl?.tryGetHostFromUrl() == request.uri.tryGetHostFromUrl()
+                val isSameDomain = lastOnLoadRequestUrl?.tryGetHostFromUrl()?.toDisplayUrl() ==
```

Comment:
Hm, I don't think we should be using `toDisplayUrl` in the engine. This is an extension function to format the URL for the toolbar, it doesn't seem right to re-use it this way. Let's refactor and add a utility/extension function to strip common subdomains and then compare the domains?

Reply:
Let's keep this check as it was and handle the special cases in the `AppLinksInteceptor`. So, we'd pass the previous URL down as well. This way the `isSameDomain` flag isn't affected by the special cases needed for app links.

Reply:
Understood, I'll move it.